aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -400,6 +416,41 @@ async def filter_constraints(
             result.append(credential)
         return result
 
+    async def local_dids_list(self):
+        """Build a local DIDs list used to verify that holder controls subject ident."""
+        async with self.profile.session() as session:
+            wallet = session.inject(BaseWallet)
+            local_did_info_list = await wallet.get_local_dids()
+            self.local_dids = []
+            for did_info in local_did_info_list:
+                if did_info.method == DIDMethod.SOV:
+                    self.local_dids.append(f"did:sov:{did_info.did}")
+                else:
+                    self.local_dids.append(did_info.did)
+
+    def field_ids_for_is_holder(self, constraints: Constraints) -> Sequence[str]:
+        """Return list of field ids for whose subject holder verification is requested."""
+        reqd_field_ids = set()
+        if not constraints.holders:
+            reqd_field_ids = []
+            return reqd_field_ids
+        for holder in constraints.holders:
+            if holder.directive == "required":
+                reqd_field_ids = set.union(reqd_field_ids, set(holder.field_ids))
+        return list(reqd_field_ids)
+
+    async def process_constraint_holders(
+        self,
+        subject_ids: Sequence[str],
+    ) -> bool:
+        """Check if holder or subject of claim still controls the identifier."""
+        if not self.local_dids:
+            await self.local_dids_list()
+        for subject_id in subject_ids:
+            if subject_id not in self.local_dids:
```

Comment:
It may be much more efficient in general to look up the relevant subject IDs in the wallet individually rather than loading every DID and then going through them.